analyzers/dataframe/JetTaggingUtils.h

```diff
@@ -20,6 +24,44 @@ namespace JetTaggingUtils{
 
   //Get flavour association of jet 
   ROOT::VecOps::RVec<int> get_flavour(ROOT::VecOps::RVec<fastjet::PseudoJet> in, ROOT::VecOps::RVec<edm4hep::MCParticleData> MCin);
+  
+  struct ghostFlavour {
+    std::vector<std::vector<float>> flavour;
+    JetClusteringUtils::FCCAnalysesJet jets;
+    std::vector<float> ghostStatus;
+    std::vector<int> MCindex;
+  };
+
+
+  //Get ghost flavour (MC flavour) of jet described here: ..
+  //a struct is returned containing
+  // - vector of flavours: <<partonFlav Jet 1, partonFlav Jet 2, ...>, <hadronFlav Jet 1, hadron Flav Jet 2, ...>>
+  // - FCCAnalysesJet struct containing resulting jets (on which functions defined in JetClusteringUtils.cc can be called)
+  // - vector of jet constituent indices <<constis jet 1>, <constis jet 2>, ...>
+  // - ghostStatus (0 = not ghost, 1 = ghost parton, 2 = ghost hadron)
+  // - MCindex vector of indices of ghosts to MC collection (and -1 if not a ghost) <part 1, ..., part n>
+  struct get_ghostFlavour {
+    get_ghostFlavour(int algo, float arg_radius, int arg_exclusive, float arg_cut, int arg_sorted, int arg_recombination, float arg_add1=0, float arg_add2=0);
+
+    int m_algo = 0; ///< flag to select jet clustering algorithm defined in JetClustering.cc (0 = kt, 1 = antikt, 2 = cambridge, 3 = eekt, 4 = ee genkt, 5 = genkt, 6 = valencia, 7 = jade) 

```

Comment:
Please move the default values to the function arguments and structure the ```struct``` like https://github.com/HEP-FCC/FCCAnalyses/blob/optmisations/analyzers/dataframe/Algorithms.h#L21#L33